### PR TITLE
[Analysis] Nullability (?. and ?:)

### PR DESCRIPTION
**Pull request by serras**
This PR adds support for:

- [X] Safe function calls `x?.fn(...)`
- [X] Checks for nullability `x == null`, `x != null`
- [x] The special operator `?:`

This is done by keeping an additional `null` uninterpreted function, which tells us whether we are in a definitely-null (if true), definitely-not-null (if false), or unknown situation.

### CODE REVIEW

**Comment by 47degdev** [comment]
Task linked: [CU-pzegv4 Handle safe calls `?.` and the Elvis operator `?:`](https://app.clickup.com/t/pzegv4)

File: plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt [hunk 1]
```diff
@@ -338,6 +361,51 @@ class LiquidDataflowTests {
...
+  fun `nullability, 1`() {
+    """
+      ${imports()}
+      fun nully1(x: Int?): Int? {
+        val y = x?.let { 1 }
+        return y.post({ it > 0 }) { "greater than 0" }
+      }
+      """(
+      // it seems that Kotlin's warning overrides ours
+      withPlugin = { failsWith { it.contains("`nully1` fails to satisfy the post-condition") } },
```
Comment (raulraja) [inline plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt:376]: I suppose this is the case related to the warnings that you mentioned in our call. I created https://app.clickup.com/t/1e0yfab so we can look into it after this PR

**Review by raulraja — APPROVED** [review]
Looking great @serras , thanks!

File: plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt (hunk 1 above)
Comment (serras) [reply plugins/liquid/refined-types-plugin/src/test/kotlin/arrow/meta/plugins/liquid/LiquidDataflowTests.kt:376]: Yes, this is one of them, the other one is `bad predicates`.

**Pull request merged**